apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,366 @@
+      if not component in result:
+        return (None, not_found_error)
+
+      result = result[component]
+
+  if result == numeric_project_id_token:
+    result = numeric_project_id
+
+  return (result, None)
+
+def convertToTextRecursive(prefix, raw_data):
+  text = u''
+  if isinstance(raw_data, list):
+    for val in raw_data:
+      text += convertToTextRecursive(prefix, val)
+  elif isinstance(raw_data, IndexedList):
+    for i in range(len(raw_data)):
+      text += convertToTextRecursive(prefix + (u'/' if len(prefix) != 0 else u'') + unicode(i),
+                                     raw_data[i])
+  elif isinstance(raw_data, OrderedDict):
+    for key in raw_data:
+      if excludeKeyInRecursiveGet(key):
+        continue
+      text += convertToTextRecursive(prefix + (u'/' if len(prefix) != 0 else u'') + key,
+                                     raw_data[key])
+  elif prefix:
+    text += prefix + u' ' + unicode(raw_data) + u'\n'
+  else:
+    text += unicode(raw_data)
+
+  return text
+
+def prepareResult(raw_data, is_recursive, is_directory, is_json, alt):
+  if is_recursive:
+    if is_json:
+      return (json.dumps(raw_data, cls=MetadataJSONEncoder, separators=(u',', u':')), None)
+    else:
+      return (convertToTextRecursive('', raw_data), None)
+
+  if is_directory and alt and not isinstance(raw_data, list):
+    return (None, bad_request_error)
+
+  if isinstance(raw_data, OrderedDict):
+    return (formatOrderedDictContents(raw_data), None)
+  
+  if isinstance(raw_data, IndexedList):
+    return (raw_data.formatContents(), None)
+  
+  if isinstance(raw_data, list):
+    if is_json:
+      return (json.dumps(raw_data, separators=(u',', u':')), None)
+    else:
+      return (''.join(s + u'\n' for s in raw_data), None)
+
+  if isinstance(raw_data, unicode) and is_json:
+    return (u'"' + raw_data + u'"', None)
+  
+  return (unicode(raw_data), None)
+
+@app.after_request
+def add_headers(response):
+  response.headers[common.server] = common.server_value
+
+  # When the token is requested, the response should not have the Metadata-Flavor header
+  if not response.data.startswith(u'{"access_token":"') and not response.data.startswith(u'access_token '):
+    response.headers[common.metadata_flavor] = common.metadata_flavor_google
+  return response
+
+@app.route('/', methods = ['GET'])
+def root():
+  return formatResponse(u'0.1/\ncomputeMetadata/\n', 200, common.content_type_text)
+
+@app.route('/computeMetadata/', methods = ['GET'])
+def computeMetadataRoot():
+  return formatResponse(u'v1/\nv1beta1/\n', 200, common.content_type_text)
+
+@app.route('/computeMetadata/v1/', methods = ['GET'])
+def getRootData():
+  return getData(u'')
+
+@app.route('/computeMetadata/v1/<path:path>', methods = ['GET'])
+def getData(path):
+  # TODO There are 500 max hanging gets (not sure if we need to enforce this)
+  # TODO Actually handle waiting for changes and last_etag
+
+  def asBool(value):
+    if not value:
```

Comment:
Yes I think that's fine. This should only get strings passed in and I just wanted to make sure we handled None, but the others can be considered False as well.